piff/kernel.py

```diff
@@ -0,0 +1,290 @@
+
+    def __repr__(self):
+        return "{0}(invLam={1!r})".format(self.__class__.__name__, self.invLam)
+
+    @property
+    def bounds(self):
+        return self._bounds
+
+
+class VonKarman(StationaryKernelMixin, NormalizedKernelMixin, Kernel):
+    """VonKarman kernel.
+
+    Parameters
+    -----------
+    length_scale : float or array with shape (n_features,), default: 1.0
+        The length scale of the kernel. If a float, an isotropic kernel is
+        used. If an array, an anisotropic kernel is used where each dimension
+        of l defines the length-scale of the respective feature dimension.
+
+    length_scale_bounds : pair of floats >= 0, default: (1e-5, 1e5)
+        The lower and upper bound on length_scale
+    """
+    def __init__(self, length_scale=1.0, length_scale_bounds=(1e-5, 1e5)):
+        self.length_scale = length_scale
+        self.length_scale_bounds = length_scale_bounds
+
+    @property
+    def anisotropic(self):
+        return np.iterable(self.length_scale) and len(self.length_scale) > 1
+
+    @property
+    def hyperparameter_length_scale(self):
+        if self.anisotropic:
+            return Hyperparameter("length_scale", "numeric",
+                                  self.length_scale_bounds,
+                                  len(self.length_scale))
+        return Hyperparameter(
+            "length_scale", "numeric", self.length_scale_bounds)
+
+    def __call__(self, X, Y=None, eval_gradient=False):
+        """Return the kernel k(X, Y) and optionally its gradient.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples_X, n_features)
+            Left argument of the returned kernel k(X, Y)
+
+        Y : array, shape (n_samples_Y, n_features), (optional, default=None)
+            Right argument of the returned kernel k(X, Y). If None, k(X, X)
+            if evaluated instead.
+
+        eval_gradient : bool (optional, default=False)
+            Determines whether the gradient with respect to the kernel
+            hyperparameter is determined. Only supported when Y is None.
+
+        Returns
+        -------
+        K : array, shape (n_samples_X, n_samples_Y)
+            Kernel k(X, Y)
+
+        K_gradient : array (opt.), shape (n_samples_X, n_samples_X, n_dims)
+            The gradient of the kernel k(X, X) with respect to the
+            hyperparameter of the kernel. Only returned when eval_gradient
+            is True.
+        """
+        from scipy.spatial.distance import pdist, cdist, squareform
+        from scipy import special
+
+        X = np.atleast_2d(X)
+        length_scale = _check_length_scale(X, self.length_scale)
+        if Y is None:
+            dists = pdist(X, metric='euclidean')
+            K = (dists**(5./6.)) * special.kv(5./6.,2*np.pi*dists/length_scale)
+            K = squareform(K)
+
+            dd_w = np.linspace(1e-4,1,100) / length_scale
+            dd = np.linspace(1e-4,1,100)
```

Comment:
I think you don't need dd, dd_w anymore.

Reply:
Yes Indeed, this is useless now. I fixed it. 

However, which is totally weird is the fact that now it does not pass Travis Cl tests for python 2.7 and 3.5 (works for python 3.4). And on my own laptop tests are working. By looking error message, it seems it is not connected with my new implementation, but with the fact that it is not able to build the environnement to start the test (fail after 2 minutes of running on Travis Cl). I tried to restart the Travis test for python 2.7 and python 3.5 few times. By the past it already happened once, and just to restart the tests worked. But it does not work now… I am not to much familiar with Travis Cl, do you have an idea of how to fixed that?

Thanks a lot

Reply:
Usually that means something changed about the Travis environment.  I'll see if I can figure it out.